Makefile

```diff
@@ -42,9 +42,15 @@ CTR_CMD            ?= $(or $(shell podman info > /dev/null 2>&1 && which podman)
 CTR_CMD_PUSH_OPTIONS ?=
 
 GENERAL_TAGS := 'include_gcs include_oss containers_image_openpgp gssapi providerless netgo osusergo libbpf '
-GPU_TAGS := ' gpu '
+GPU_TAGS := ' '
+ifeq ($(shell ldconfig -p | grep -q libnvml_injection.so && echo exists),exists)
+	GPU_TAGS := ' nvml '
+endif
+ifeq ($(shell ldconfig -p | grep -q libdcgm.so && echo exists),exists)
+	GPU_TAGS := ' dcgm '
+endif
 ifeq ($(shell ldconfig -p | grep -q libhlml.so && echo exists),exists)
```

Comment:
We are only prioritising dcgm over NVML. 

Dcgm/nvml won't exist in the same system

Reply:
nothing actually prevents all 3 shared libraries being on the same machine, so that's really beyond our control. However if the build machine has all 3 shared libraries, then the makefile will choose `habana` over `dcgm` over `nvml` - right?

Reply:
Someone could install all the libraries on the system I guess. I think it would be rather unconventional as it's unlikely that you would find an NVIDIA GPU plugged into a system with a Habana CPU. 
If we think we need to cater for all three we can make the build flag build up all the devices together